examples/chat-online-status/react/src/app/page.tsx

```diff
@@ -0,0 +1,95 @@
+"use client";
+
+import React, { useEffect, useState } from 'react';
+import { useChatClient, useRoom, usePresenceListener, usePresence } from '@ably/chat/react';
+import { PresenceEvent } from '@ably/chat';
+import '../../styles/styles.css';
+
+interface OnlineStatus {
+  status: string;
+}
+
+export default function Home() {
+  const { roomStatus, connectionStatus } = useRoom();
+
+  if (roomStatus !== 'attached' || connectionStatus !== 'connected') {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div>
+      <Online />
+    </div>
+  );
+}
+
+const Online = () => {
+  const [onlineStatuses, setOnlineStatuses] = useState<PresenceEvent[]>([]);
+  const { presenceData, error } = usePresenceListener({
+    listener: (presence) => {
+      if (presence.action === 'enter') {
+        setOnlineStatuses((prevStatus: PresenceEvent[]) => [...prevStatus, presence as PresenceEvent]);
+      } else if (presence.action === 'leave') {
+        setOnlineStatuses((prevStatus: PresenceEvent[]) => prevStatus.filter((status: PresenceEvent) => status.clientId !== presence.clientId));
```

Comment:
Does `presence.action === 'update'` need to be handled here? Also, if so, could we turn this into a switch? :)